src/tainting/Dataflow_tainting.ml

```diff
@@ -168,8 +171,15 @@ module Top_sinks = struct
         if r'.start > r.start || r'.end_ < r.end_ then
           (* The new match is a worse fit so we keep the current one. *)
           sinks
-        else (* We found a better (larger) match! *)
-          S.add m' (S.remove m sinks)
+        else
+          (* We found a better (larger) match! *)
+          (* There may be several matches in `sinks` that are subsumed by `m'`.
+           * E.g. we could have sinks (1,5) and (5,10), and then add `(1,10)`
```

Comment:
i think by this you mean a match with range (1, 5) and (5, 10). could you just clarify this a little more? i was a little confused when reading this at first